
package SwitchAdapter_Proto;

option java_package = "zsdn.protocol";
option java_outer_classname = "SwitchAdapterProtocol";

// All Requests the SwitchAdapter will handle

message Request  {

    oneof RequestMsg {
        HandleOpenflowMessagesRequest handle_openflow_messages_request = 1;
    }

    // Request submessage definitions

    message HandleOpenflowMessagesRequest {
        repeated bytes open_flow_message = 1;
    }
}





// All replies the SwitchAdapter can send

message Reply {

    oneof ReplyMsg {
        HandleOpenflowMessagesReply handle_openflow_messages_reply = 1;
    }

    // Reply submessage definitions

    message HandleOpenflowMessagesReply {}
}
